src/type-checker.cc

```diff
@@ -172,10 +172,10 @@ Result TypeChecker::GetThisFunctionLabel(Label** label) {
 
 Result TypeChecker::PeekType(Index depth, Type* out_type) {
   Label* label;
+  *out_type = Type::Any;
```

Comment:
Why is this needed?

Reply:
One of the callsites of PeekType (TypeChecker::OnIndexedFuncRef) assumes that it will initialize the `Type*` argument even in case of an `Error` result, but if the `Error` is because of a failure in the `CHECK_RESULT(TopLabel(&label));` line, that won't happen. Moving this line higher makes sure that the `Type*` gets initialized even in that kind of error.

Reply:
Isn't the callsite that is wrong then?  I don't think we should need to write to the output variable at all if we are returning an error

Reply:
OK, let me revisit this and see how that ends up looking.

Reply:
Here's an attempt to fix that callsite instead (which now adds some additional logged error messages to two tests).